### PR TITLE
test: switching to stubserver in tests instead of testservice 

### PR DESCRIPTION
**Pull request by pvsravani**
Partially Addresses: #7291

RELEASE NOTES: None

### CODE REVIEW

**Review by purnesh42H — APPROVED** [review]
lgtm

**Comment by purnesh42H** [comment]
@easwars for second review

File: test/balancer_test.go [hunk 1]
```diff
@@ -417,11 +417,17 @@ func (s) TestAddressAttributesInNewSubConn(t *testing.T) {
...
 		t.Fatal(err)
 	}
+	stub := &stubserver.StubServer{
+		Listener: lis,
+		EmptyCallF: func(_ context.Context, _ *testpb.Empty) (*testpb.Empty, error) {
+			return &testpb.Empty{}, nil
+		},
+	}
+
+	stub.S = grpc.NewServer()
```
Comment (easwars) [inline test/balancer_test.go:427]: How about moving the initialization of the `S` field inside the literal struct defintion?

```golang
	stub := &stubserver.StubServer{
		Listener: lis,
		EmptyCallF: func(_ context.Context, _ *testpb.Empty) (*testpb.Empty, error) {
			return &testpb.Empty{}, nil
		},
		S: grpc.NewServer(),
	}
```

Here and in the other three places in this PR.

Comment (pvsravani) [reply test/balancer_test.go]: Done

**Review by easwars — APPROVED** [review]

**Pull request merged**